### DISCUSSION TITLE
Configurazione plancia energetica

### DISCUSSION THREAD

**Discussion by Borghese** [indomus/forum#2774] [GENERALE - Discussioni generiche]
Ciao a tutti,

dopo aver installato uno shelly em con la pinza che misura il consumo di casa sulla fase del magnetotermico differenziale nel quadro elettroci (a valle del contatore di scambio quindi) oggi ho finalmente installato un altro shelly em, con la pinza che misura la produzione del FV, inserendola nella fase tra inverter e contatore di produzione.

Ora stavo provando a configurare sia l'app shelly che home assistant, per capire in ordine :

Quanto consumo
Quanto Produco
Quanto imetto in rete

ma ho un pò di difficoltà, nel senso che :

Con la pinza a valle del contatore di scambio leggo perfettamente quanto sto consumando -- > Questo sensore è inserito nella sezione rete elettrica di home assistant
Con la pinza tra inverter e contatore di produzione leggo esattamente quando sto producendo -- > Questo sensore è inserito nella sezione Pannelli Solari di home assistant

Non mi è per nulla chiaro come faccio a configurarlo in modo che se l'impianto produce 2kwh, e io ne sto consumando per es 1kwh, il resto vada in rete.

Cosa ci devo mettere nella sezione "Restituito alla Rete" di home assistant ?

Perchè ho fatto un pò di prove, ma se ci metto la produzione del FV, l'energia prodotta mi segna sempre che la mette in rete, praticamente non autoconsumo nulla (foto1)
<a href="https://imgbb.com/"><img src="https://i.ibb.co/FhvtsjT/1.png" alt="1" border="0"></a>

Se invece metto il sensore energy returned mi segna sempre zero, e nei consumo dell'abitazione mi somma il prelievo dalla rete e la produzione del FV (foto2)
<a href="https://imgbb.com/"><img src="https://i.ibb.co/SBtRtBV/2.png" alt="2" border="0"></a>

mi aiutate a capire ?

questa è la configurazione della plancia : 
<a href="https://ibb.co/nLgL9Fj"><img src="https://i.ibb.co/ZMLM50x/3.png" alt="3" border="0"></a>

grazie in anticipo a chiunque mi potrà aiutare...

**Comment by crug80** [comment]
Premetto che non conosco i componenti della shelly, ma credo che concettualmente il ragionamento sia uguale ad altri sistemi. 
Lungo la linea del contatore di scambio passano sia i prelievi che le immissioni, quindi tu dovresti vedere valori positivi nel caso di prelievi dalla rete, mentre valori negativi nel caso di immissioni. Se ho capito bene come funziona, nell'immissione in rete dovresti mettere l'energy returned della pinza che hai a valle del contatore di scambio. 

Non hai pensato, invece, di integrare direttamente l'inverter?

**Reply by Borghese** [reply]
grazie Crug80 per la risposta.
sto iniziando a pensare di aver posizionato la pinza dei prelievi in un posto sbagliato
Questa è come è ora : 
<a href="https://ibb.co/m8VSm8W"><img src="https://i.ibb.co/Qd3XZdw/1111.png" alt="1111" border="0"></a>

ma probabilmente il punto corretto è prima del quadro generale di casa, direttamente sulla fase che esce dal contatore di scambio, e dove passano appunto, come dici tu, sia i prelievi che le immissioni.
Lunedì provo a spostare quello shelly.

A integrare direttamente l'inverter si ci ho pensato, anche se io ho un aurora power one 3.0 e in rete non ho trovato grande materiale, forse con seriale ci si potrebbe riuscire... dici che sarebbe meglio come soluzione ?

**Comment by crug80** [comment]
Il mio impianto è differente, poiché ho un inverter ibrido con batterie di accumulo e linea EPS, pertanto per avere le info corrette, ovvero anche la porzione di energia che va e viene da e verso le batterie ragionare solo in termini di TA non mi era sufficiente. Nel tuo caso sì, dovrebbe esserlo, ma devi secondo me aggiungere una clamp dopo il quadro generale e prima del contatore enel, non spostare quello che hai.  Se sposti quello che hai, ti perdi l'auto consumo di casa. E' vero anche che questo dato potresti ricavartelo, poiché il consumo è dato dalla differenza (positiva/negativa) tra quello che passa e quello che produci. Fai una prova prima di spendere altri soldi.  
Nel tuo caso, non avendo inverter ibrido, credo che interfacciare l'inverter non ti darebbe informazioni in più. Se hai contatori nuovi di tipo Open Meter potresti anche valutare l'opzione Alfa sinapsi il cui componente è sia integrabile in HA che di più semplice installazione. E' addirittura in grado di avvisarti quando l'enel è in procinto di staccarti per troppo consumo. Cmq, visto che hai già speso soldi per shelly e clamp, a questo punto cerca di far lavorare  bene quelli.

**Comment by Borghese** [comment]
Ho risolto ragazzi !

Ho spostato lo shelly con la pinza che monitorava i consumi di casa, a valle del contatore di scambio, mettendo la pinza sulla fase appunto in uscita del contatore di scambio.
Ora mi contabilizza correttamente energia positiva se prelievo, e negativa se immetto

vi allego la foto della situazione attuale : (con la pdc dell'acs in piena produzione )

<a href="https://imgbb.com/"><img src="https://i.ibb.co/YP0B1bF/2222.png" alt="2222" border="0"></a>

**Comment by crug80** [comment]
quindi la sottrazione che dicevo prima te la sta facendo la dashboard..  :) 
Perfect!

**Comment by 90romario** [comment]
Ciao, potresti gentilmente postare i vari sensori inseriti nella plancia per avere la stessa dashboard energia? mi ritrovo nella tua stessa situazione e non riesco a venirne a capo grazie

**Reply by Borghese** [reply]
ciao,
dimmi se ti aiuta così, ti riporto direttamente il codice :

type: custom:power-flow-card-plus
entities:
  grid:
    entity: sensor.produzione_fv_power
    color_icon: consumption
    display_zero_tolerance: 0
    display_state: two_way
    invert_state: false
    use_metadata: false
    color_value: false
    secondary_info: {}
  solar:
    entity: sensor.produzione_fv_channel_2_power
    invert_state: false
    use_metadata: false
    display_zero_state: true
    color_icon: true
    color_value: false
  home:
    override_state: false
    subtract_individual: false
  individual1:
    entity: sensor.scaldabagno_pdc_power
    name: Pdc Acs
    icon: mdi:water-boiler
    color_value: false
    color_icon: true
    display_zero: false
    display_zero_state: false
    calculate_flow_rate: false
    show_direction: false
    inverted_animation: false
    use_metadata: false
  individual2:
    entity: sensor.lavasciuga_power
    name: Lavasciuga
    icon: mdi:washing-machine
    color_icon: true
    show_direction: false
clickable_entities: true
display_zero_lines:
  mode: show
  transparency: 50
  grey_color:
    - 189
    - 189
    - 189
use_new_flow_rate_model: true
w_decimals: 0
kw_decimals: 1
min_flow_rate: 0.75
max_flow_rate: 6
max_expected_power: 2000
min_expected_power: 0.01
watt_threshold: 1000
transparency_zero_lines: 0
title: Dashboard Energia

**Comment by 90romario** [comment]
Ciao, guarda io ho uno shelly em con due pinze, 1 a monte del quadro elettrico e l'altra all'entrata della fase del contatore di produzione ( in quanto il contatore di produzione è lontano dal quadro) Il problema è che quando inserisco le entita nel pannello energia mi da valori totalmente contrastanti. Essendo ancora non pratico di HA questo codice dove dovrei inserirlo? scusami tanto se ne approfitto

**Reply by Borghese** [reply]
ok allora ho mal interpretato la tua domanda.
Se il tuo obiettivo è quello di avere la dashboard energia nativa di HA fai cosi :

Inizia a verificare se il valore misurato dalle pinze che hai posizionato, corrisponde a quanto prodotto dall'inverter e quanto segnalato dal contatore di prelievo per i consumi
Giusto per capire se sono posizionate correttamente, sia come posizione rispetto all'impianto, sia nel verso
Poi è sufficiente aggiungere le entità su HA nella sezione IMPOSTAZIONI/PLANCE/ENERGIA nei tab di RETE ELETTRICA e PANNELLI SOLARI
Per quanto riguarda lo schema di installazioen delle pinze, prova a dare uno sguardo a questa pagina, ci sono gli schemi a seconda di come è configurato il tuo impianto, dovrebbe essere abbastanza intuitivo : https://www.energeticambiente.it/fonti-di-energia-rinnovabile/fotovoltaico/tecnica-componentistica-e-installazione/80606-monitoraggio-impianto-fotovoltaico-con-shelly-em/page13
(spero di non violare alcun regolamento postando pagina di un altro forum)

**Comment by 90romario** [comment]
No credo tu abbia capito bene la domanda mi sono spiegato male io, per quanto riguarda il posizionamento delle pinze ok so come fare, volevo impostare la tua stessa plancia energiaIl 4 Apr 2024 09:59, Borghese ***@***.***> ha scritto:
ok allora ho mal interpretato la tua domanda.
Se il tuo obiettivo è quello di avere la dashboard energia nativa di HA fai cosi :
Inizia a verificare se il valore misurato dalle pinze che hai posizionato, corrisponde a quanto prodotto dall'inverter e quanto segnalato dal contatore di prelievo per i consumi
Giusto per capire se sono posizionate correttamente, sia come posizione rispetto all'impianto, sia nel verso
Poi è sufficiente aggiungere le entità su HA nella sezione IMPOSTAZIONI/PLANCE/ENERGIA nei tab di RETE ELETTRICA e PANNELLI SOLARI
Per quanto riguarda lo schema di installazioen delle pinze, prova a dare uno sguardo a questa pagina, ci sono gli schemi a seconda di come è configurato il tuo impianto, dovrebbe essere abbastanza intuitivo : https://www.energeticambiente.it/fonti-di-energia-rinnovabile/fotovoltaico/tecnica-componentistica-e-installazione/80606-monitoraggio-impianto-fotovoltaico-con-shelly-em/page13
(spero di non violare alcun regolamento postando pagina di un altro forum)

**Comment by 90romario** [comment]
No credo tu abbia capito bene la domanda mi sono spiegato male io, per quanto riguarda il posizionamento delle pinze ok so come fare, volevo impostare la tua stessa plancia energia

**Reply by Borghese** [reply]
ok capito!
In questo caso dai uno sguardo a questi link : 

https://community.home-assistant.io/t/power-flow-card-plus/552326
https://www.vincenzocaputo.com/home_assistant/power-flow-card-plus-in-home-assistant-facilissimo-919
https://github.com/flixlix/power-flow-card-plus

In buona sostanza è una card apposita, e una volta installata, puoi aggiungere i singoli componenti di misurazione, a seconda del loro impiego.
Su GRID per es ci andrà la pinza che misura il prelievo
su SOLAR quello di produzione
BATTERY se hai accumulo 
INDIVIDUAL 1 e 2 puoi aggiungere pinze (es shelly PM) che misurano assorbimento di singoli dispositivi

nei link sopra dovresti trovare tutte le risposte!
